src/sagemaker/pipeline.py

```diff
@@ -156,6 +157,9 @@ def deploy(
             data_capture_config (sagemaker.model_monitor.DataCaptureConfig): Specifies
                 configuration related to Endpoint data capture for use with
                 Amazon SageMaker Model Monitoring. Default: None.
+            kms_key (str): The ARN of the KMS key that is used to encrypt the
```

Comment:
I think this can actually be the arn, key id or alias - https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_OutputDataConfig.html